### ISSUE TITLE
Valid BigQuery ARRAY_ARG syntax can't be parsed by sqlfluff, throws error

### ISSUE THREAD

**Issue by jamiet-msm** [sqlfluff/sqlfluff#1171]
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

## Expected Behaviour

Valid BigQuery syntax should be parsed successfully

## Observed Behaviour

sqlfluff throws an error:

```bash
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                select_clause_element:
[L:  2, P:  5]      |                    wildcard_expression:
[L:  2, P:  5]      |                        wildcard_identifier:
[L:  2, P:  5]      |                            star:                             '*'
[L:  2, P:  6]      |            whitespace:                                       ' '
[L:  2, P:  7]      |            newline:                                          '\n'
[L:  3, P:  1]      |            [META] dedent:
[L:  3, P:  1]      |            from_clause:
[L:  3, P:  1]      |                keyword:                                      'FROM'
[L:  3, P:  5]      |                whitespace:                                   ' '
[L:  3, P:  6]      |                newline:                                      '\n'
[L:  4, P:  1]      |                whitespace:                                   '    '
[L:  4, P:  5]      |                from_expression:
[L:  4, P:  5]      |                    [META] indent:
[L:  4, P:  5]      |                    from_expression_element:
[L:  4, P:  5]      |                        table_expression:
[L:  4, P:  5]      |                            bracketed:
[L:  4, P:  5]      |                                start_bracket:                '('
[L:  4, P:  6]      |                                [META] indent:
[L:  4, P:  6]      |                                newline:                      '\n'
[L:  5, P:  1]      |                                whitespace:                   '        '
[L:  5, P:  9]      |                                select_statement:
[L:  5, P:  9]      |                                    select_clause:
[L:  5, P:  9]      |                                        keyword:              'select'
[L:  5, P: 15]      |                                        [META] indent:
[L:  5, P: 15]      |                                        newline:              '\n'
[L:  6, P:  1]      |                                        whitespace:           '            '
[L:  6, P: 13]      |                                        select_clause_element:
[L:  6, P: 13]      |                                            expression:
[L:  6, P: 13]      |                                                function:
[L:  6, P: 13]      |                                                    function_name:
[L:  6, P: 13]      |                                                        function_name_identifier:  'array_agg'
[L:  6, P: 22]      |                                                    bracketed:
[L:  6, P: 22]      |                                                        start_bracket:  '('
[L:  6, P: 23]      |                                                        [META] indent:
[L:  6, P: 23]      |                                                        expression:
[L:  6, P: 23]      |                                                            column_reference:
[L:  6, P: 23]      |                                                                identifier:  't'
[L:  6, P: 24]      |                                                        whitespace:  ' '
[L:  6, P: 25]      |                                                        orderby_clause:
[L:  6, P: 25]      |                                                            keyword:  'order'
[L:  6, P: 30]      |                                                            whitespace:  ' '
[L:  6, P: 31]      |                                                            keyword:  'by'
[L:  6, P: 33]      |                                                            [META] indent:
[L:  6, P: 33]      |                                                            whitespace:  ' '
[L:  6, P: 34]      |                                                            column_reference:
[L:  6, P: 34]      |                                                                identifier:  't'
[L:  6, P: 35]      |                                                                dot:  '.'
[L:  6, P: 36]      |                                                                identifier:  'timestampCreated'
[L:  6, P: 52]      |                                                            whitespace:  ' '
[L:  6, P: 53]      |                                                            keyword:  'desc'
[L:  6, P: 57]      |                                                            [META] dedent:
[L:  6, P: 57]      |                                                        whitespace:  ' '
[L:  6, P: 58]      |                                                        expression:
[L:  6, P: 58]      |                                                            column_reference:
[L:  6, P: 58]      |                                                                identifier:  'limit'
[L:  6, P: 63]      |                                                        whitespace:  ' '
[L:  6, P: 64]      |                                                        expression:
[L:  6, P: 64]      |                                                            literal:  '1'
[L:  6, P: 65]      |                                                        [META] dedent:
[L:  6, P: 65]      |                                                        end_bracket:  ')'
[L:  6, P: 66]      |                                                array_accessor:
[L:  6, P: 66]      |                                                    start_square_bracket:  '['
[L:  6, P: 67]      |                                                    [META] indent:
[L:  6, P: 67]      |                                                    expression:
[L:  6, P: 67]      |                                                        function:
[L:  6, P: 67]      |                                                            function_name:
[L:  6, P: 67]      |                                                                function_name_identifier:  'offset'
[L:  6, P: 73]      |                                                            bracketed:
[L:  6, P: 73]      |                                                                start_bracket:  '('
[L:  6, P: 74]      |                                                                [META] indent:
[L:  6, P: 74]      |                                                                expression:
[L:  6, P: 74]      |                                                                    literal:  '0'
[L:  6, P: 75]      |                                                                [META] dedent:
[L:  6, P: 75]      |                                                                end_bracket:  ')'
[L:  6, P: 76]      |                                                    [META] dedent:
[L:  6, P: 76]      |                                                    end_square_bracket:  ']'
[L:  6, P: 77]      |                                            unparsable:       !! Expected: 'Nothing...'
[L:  6, P: 77]      |                                                raw:          '.'
[L:  6, P: 78]      |                                                raw:          '*'
[L:  6, P: 79]      |                                    newline:                  '\n'
[L:  7, P:  1]      |                                    whitespace:               '        '
[L:  7, P:  9]      |                                    [META] dedent:
[L:  7, P:  9]      |                                    from_clause:
[L:  7, P:  9]      |                                        keyword:              'from'
[L:  7, P: 13]      |                                        newline:              '\n'
[L:  8, P:  1]      |                                        whitespace:           '            '
[L:  8, P: 13]      |                                        from_expression:
[L:  8, P: 13]      |                                            [META] indent:
[L:  8, P: 13]      |                                            from_expression_element:
[L:  8, P: 13]      |                                                table_expression:
[L:  8, P: 13]      |                                                    table_reference:
[L:  8, P: 13]      |                                                        identifier:  '`myproject.mydataset.mytable*`'
[L:  8, P: 43]      |                                                whitespace:   ' '
[L:  8, P: 44]      |                                                alias_expression:
[L:  8, P: 44]      |                                                    identifier:  't'
[L:  8, P: 45]      |                                            [META] dedent:
[L:  8, P: 45]      |                                    newline:                  '\n'
[L:  9, P:  1]      |                                    whitespace:               '        '
[L:  9, P:  9]      |                                    groupby_clause:
[L:  9, P:  9]      |                                        keyword:              'group'
[L:  9, P: 14]      |                                        whitespace:           ' '
[L:  9, P: 15]      |                                        keyword:              'by'
[L:  9, P: 17]      |                                        [META] indent:
[L:  9, P: 17]      |                                        newline:              '\n'
[L: 10, P:  1]      |                                        whitespace:           '            '
[L: 10, P: 13]      |                                        column_reference:
[L: 10, P: 13]      |                                            identifier:       't'
[L: 10, P: 14]      |                                            dot:              '.'
[L: 10, P: 15]      |                                            identifier:       'UserAgent'
[L: 10, P: 24]      |                                        [META] dedent:
[L: 10, P: 24]      |                                newline:                      '\n'
[L: 11, P:  1]      |                                whitespace:                   '    '
[L: 11, P:  5]      |                                [META] dedent:
[L: 11, P:  5]      |                                end_bracket:                  ')'
[L: 11, P:  6]      |                        whitespace:                           ' '
[L: 11, P:  7]      |                        alias_expression:
[L: 11, P:  7]      |                            keyword:                          'AS'
[L: 11, P:  9]      |                            whitespace:                       ' '
[L: 11, P: 10]      |                            identifier:                       'TBL'
[L: 11, P: 13]      |                    [META] dedent:

==== parsing violations ====
L:   6 | P:  77 |  PRS | Line 6, Position 77: Found unparsable section: '.*'
```

## Steps to Reproduce

1. Write the following to file.sql:
```sql
SELECT 
    * 
FROM 
    (
        select
            array_agg(t order by t.timestampCreated desc limit 1)[offset(0)].*
        from
            `myproject.mydataset.mytable*` t
        group by
            t.UserAgent
    ) AS TBL
```

2. `sqlfluff parse file.sql --dialect bigquery`

## Version
sqlfluff, version 0.6.0
Python 3.8.10

I'm sympathetic to you folks because I love what you're doing here with sqlfluff and this is definitely an edge case. In fact, if I strip off the problematic asterisk from the end of line 6 then it parses successfully and the logical query hasn't changed at all (i.e. produces the same result).

Nevertheless, this is a valid BigQuery statement, written in good faith by one of my colleagues, and sqlfluff wasn't able to parse it. I thought you'd want to know.

**Comment by barrywhart** [comment]
@tunetheweb: This seems similar to your recent work on BigQuery functions returning structures (#1276). Any thoughts on this one?

**Comment by tunetheweb** [comment]
Looks like it lints to me now thanks to #1269. So think this can be closed.

**Comment by barrywhart** [comment]
@tunetheweb: Can you try running `sqlfluff parse` on the file? That fails for me. Note that `sqlfluff lint` will still lint the file, even with parse errors.

My lint output includes the following line (easily overlooked, because it looks like just another lint error):
```
L:   6 | P:  77 |  PRS | Line 6, Position 77: Found unparsable section: '.*'
```

**Comment by tunetheweb** [comment]
Looks like I was linting my other test file! :-)

I see the error now. Will have a look as should be easy enough to support this.

**Comment by barrywhart** [comment]
If you are able to do it, that'd be great. I was just reviewing some of the recent issues today and saw the similarity.

**Comment by tunetheweb** [comment]
Suggested fix raised in https://github.com/sqlfluff/sqlfluff/pull/1278.

**Comment by jamiet-msm** [comment]
Thank you to @tunetheweb & @barrywhart . I'm not familiar with sqlfluff's release cadence, do you know if/when this will make it into a release that I can pip install?

**Comment by tunetheweb** [comment]
The [GitHub home page says](https://github.com/sqlfluff/sqlfluff#releases):

> New releases are made monthly. For more information, visit [Releases](https://github.com/sqlfluff/sqlfluff/releases).

and it’s been 25 days since last release, so hopefully soon.

@barrywhart id be keen on a release soon too as [there’s lots of good changes committed to main](https://github.com/sqlfluff/sqlfluff/blob/main/CHANGELOG.md).

**Comment by barrywhart** [comment]
Releases take a bit of manual work but are pretty easy. I'll go ahead and work on creating one some time this week.

**Comment by jamiet-msm** [comment]
> @jamiet-msm, @tunetheweb: I just released 0.6.3.
> 
> https://github.com/sqlfluff/sqlfluff/releases/tag/0.6.3
> 
> I have confirmed it's up on PyPI as well.

Brilliant. Thank you @barrywhart
